frontend/packages/dev-console/src/components/gitops/GitOpsList.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import GitOpsListItem from './GitOpsListItem';
+import {
+  Stack,
+  StackItem,
+  Split,
+  SplitItem,
+  EmptyState,
+  EmptyStateVariant,
+  EmptyStateIcon,
+} from '@patternfly/react-core';
+import { getImageForIconClass } from '@console/internal/components/catalog/catalog-item-icon';
+import { GitOpsAppGroupData } from './gitops-types';
+import './GitOpsList.scss';
+
+interface GitOpsListProps {
+  appGroups: GitOpsAppGroupData[];
+  emptyStateMsg: string;
+}
+
+const GitOpsList: React.FC<GitOpsListProps> = ({ appGroups, emptyStateMsg }) => {
+  const emptyState = () => {
+    const gitopsImage = () => (
+      <img
+        className="odc-gitops-list__empty-state__icon"
+        src={getImageForIconClass('icon-gitops')}
+        alt="GitOps icon"
+      />
+    );
+    return (
+      <EmptyState variant={EmptyStateVariant.full}>
+        <p className="odc-gitops-list__empty-state__msg">{emptyStateMsg}</p>
+        <EmptyStateIcon variant="container" component={gitopsImage} />
+      </EmptyState>
+    );
+  };
+  return (
+    <div className="odc-gitops-list">
+      {!emptyStateMsg ? (
+        <Stack hasGutter>
+          <StackItem>
+            <Split>
+              <SplitItem isFilled />
+              <SplitItem>{`${_.size(appGroups)} items`}</SplitItem>
+            </Split>
+          </StackItem>
+          {_.map(appGroups, (appGroup) => (
+            <StackItem>
+              <GitOpsListItem appGroup={appGroup} />
+            </StackItem>
+          ))}
+        </Stack>
+      ) : (
+        emptyState()
```

Comment:
Once you fix the above comment on `gitopsImage` you can inline the JSX here instead.